go/src/hello_test.go

```diff
@@ -0,0 +1,21 @@
+package greetings
+
+import (
+    "testing"
+    "regexp"
+)
+
+
+
+// TestHelloName calls greetings.Hello with a name, checking
+// for a valid return value.
+func TestHelloName(t *testing.T) {
+    name := "Gladys"
+    want := regexp.MustCompile(`\b`+name+`\b`)
+    msg, err := Hello("Gladysx")
```

Comment:
What actually is this test doing. How does the random element above impact passing logic

Reply:
Right now this isn't random. I just wanted to have some test that I could upload results from. For each of our examples should it randomly fail? What % of the time?

Reply:
I guess I saw usage of random and expected some failure expectation - 

I think it's fine to have tests that pass - I just wasn't sure what expectation was of this function and what test was checking really.